addons/website_quote/models/order.py

```diff
@@ -405,12 +405,19 @@ def on_change_product_id(self, cr, uid, ids, product, uom_id=None, context=None)
             domain = {'uom_id': [('category_id', '=', product_obj.uom_id.category_id.id)]}
         return {'value': vals, 'domain': domain}
 
+    # TODO master: to remove, replaced by onchange of the new api
     def product_uom_change(self, cr, uid, ids, product, uom_id, context=None):
         context = context or {}
         if not uom_id:
             return {'value': {'price_unit': 0.0, 'uom_id': False}}
         return self.on_change_product_id(cr, uid, ids, product, uom_id=uom_id, context=context)
 
+    @api.onchange('uom_id')
+    def _product_uom_change(self):
+        if not self.uom_id:
+            self.price_unit = 0.0
+        return self._onchange_product_id()
+
     @api.onchange('product_id')
```

Comment:
You get rid of `_product_uom_change` and use `@api.onchange('product_id', 'uom_id')`